include/libopencm3/pac55xx/memctl.h

```diff
@@ -0,0 +1,114 @@
+/**
+ * @brief Memory Controller definitions for the Qorvo PAC55xx series of microcontrollers
+ *
+ * @addtogroup PAC55xx_memctl MEMCTL
+ * @ingroup PAC55xx_defines
+ * @author Kevin Stefanik <kevin@allocor.tech>
+ * LGPL License Terms @ref lgpl_license
+ * @date 17 Mar 2020
+ *
+ * Definitions in this file come from the PAC55XX Family User Guide Rev 1.23
+ * by Active-Semi dated November 19, 2019.
+ */
+/*
+ * This file is part of the libopencm3 project.
+ *
+ * Copyright (C) 2020 Kevin Stefanik <kevin@allocor.tech>
+ *
+ * This library is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This library is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this library.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#ifndef LIBOPENCM3_PAC55XX_MEMCTL_H_
+#define LIBOPENCM3_PAC55XX_MEMCTL_H_
+
+#include <libopencm3/cm3/common.h>
+#include <libopencm3/pac55xx/memorymap.h>
+/**@{*/
+
+
+/** @defgroup memctl_reg Memory Controller Configuration Register
+@{*/
+/** Memory Controller Configuration Register */
+#define MEMCTL          MMIO32(MEMCTL_BASE)
+#define MEMCTL_WSTATE_MASK              (0xF)
+#define MEMCTL_WSTATE(ws)               ((ws) & MEMCTL_WSTATE_MASK)
+#define MEMCTL_MCLKDIV_MASK             (0xF)
+#define MEMCTL_MCLKDIV_SHIFT            4
+#define MEMCTL_MCLKDIV(d)               (((d) & MEMCTL_MCLKDIV_MASK) << MEMCTL_MCLKDIV_SHIFT)
+#define MEMCTL_MCLKDIV1     (0)
+#define MEMCTL_MCLKDIV2     (1)
+#define MEMCTL_MCLKDIV3     (2)
+#define MEMCTL_MCLKDIV4     (3)
+#define MEMCTL_MCLKDIV5     (4)
+#define MEMCTL_MCLKDIV6     (5)
+#define MEMCTL_MCLKDIV7     (6)
+#define MEMCTL_MCLKDIV8     (7)
+#define MEMCTL_MCLKDIV9     (8)
+#define MEMCTL_MCLKDIV10    (9)
+#define MEMCTL_MCLKDIV11    (10)
+#define MEMCTL_MCLKDIV12    (11)
+#define MEMCTL_MCLKDIV13    (12)
+#define MEMCTL_MCLKDIV14    (13)
+#define MEMCTL_MCLKDIV15    (14)
+#define MEMCTL_MCLKDIV16    (15)
```

Comment:
this looks like it would be easier and simpler and clearere for the users to be just
```
#define MEMCTL_MCLKDIV(d) (((d-1) & MASK << SHIFT)
```
This lets users jsut do ```MEMCTL |= MEMCTL_MCLKDIV(4)``` instead of ```MEMCTL |= MEMCTL_MCLKDIV(MEMCTL_MCLKDIV4)``` and that seems simpler and easier to read and work with.

Reply:
done